### DISCUSSION TITLE
Missing 'putty' executable after installation via Homebrew on macOSGetting Started

### DISCUSSION THREAD

**Discussion by abidmaria** [Homebrew/discussions#4945] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.22
ORIGIN: https://github.com/Homebrew/brew

HEAD: 251a098fbf1c02ca95f53b4806e5b6da8d5e4a80

Last commit: 7 days ago

Core tap JSON: 02 Dec 13:32 UTC

HOMEBREW_PREFIX: /usr/local

HOMEBREW_CASK_OPTS: []

HOMEBREW_MAKE_JOBS: 4

Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby

CPU: quad-core 64-bit icelake

Clang: 15.0.0 build 1500

Git: 2.39.3 => /Library/Developer/Comma
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

**Environment:**
- macOS Version: 13.6.1
- Homebrew Version: 4.1.22
**Issue:**
After installing PuTTY via Homebrew (`brew install putty`), the main `putty` executable is not available. The installation process completes without errors, and other PuTTY-related executables like `plink`, `pscp`, `psftp`, `psusan`, and `puttygen` are installed and symlinked correctly in `/usr/local/bin`. However, the `putty` executable itself is missing, resulting in a `command not found` error when trying to run `putty` in the terminal.

**Steps to Reproduce:**
1. Run `brew install putty`
2. After installation, check for the `putty` executable using `ls -l /usr/local/Cellar/putty/0.79/bin`
3. Try to run `putty` in the terminal.

**Expected Behavior:**
The `putty` command should be recognized and executable after installation.

**Actual Behavior:**
The `putty` executable is missing, and the command is not recognized in the terminal.

Any assistance or insights into this issue would be greatly appreciated.

**Comment by osalbahr** [comment]
The formula does not provide a `putty` executable, as you mentioned:
```console
% ls $(brew --cellar putty)/*/bin
pageant		pscp		psusan
plink		psftp		puttygen
```
Could you provide a link to the documentation showing that a `putty` executable is expected?

**Comment by cho-m** [comment, marked as answer]
The formula currently only builds the command line tools and not the GTK+3 GUI.

This was a decision made when Homebrew dropped support for options. Removing GUI support was based on analytics data from users (though only HEAD had support at that time) - https://github.com/Homebrew/homebrew-core/commit/4efd8c5ee79c5cd7662876ece92a58035ae66a6b

This would be considered a feature request, which is usually best handled by opening a pull request with desired changes to formula.

**Comment by abidmaria** [comment]
Thank you for your response and the clarification regarding the PuTTY formula in Homebrew. I understand now that the formula includes only the command-line tools and not the GTK+3 GUI, based on the decision made when options were dropped from Homebrew formulas.

Considering this, I believe it might be helpful for users, especially those less familiar with Homebrew or migrating from Windows, if this information were more prominently documented. This could potentially prevent confusion and enhance user experience. 

Would it be feasible to include a note in the formula's documentation or on the Homebrew website to clarify this aspect of the PuTTY package for future users?

Thank you again for your time and consideration.

**Comment by Devon7** [comment]
Must I install something called `GTK+3`on MacOS to get PuTTY?
If so, the build process should **prominently** say so!

```
$ git clone https://git.tartarus.org/simon/putty.git
$ cd putty
$ cmake .
$ cmake --build .
$ find . -type f -perm +1 -iname '*putty*'
./puttygen

```

**Reply by gromgit** [reply]
Why is this a Homebrew issue, when you're manually building PuTTY from upstream sources without using `brew install`?

**Comment by FrenzyExists** [comment]
This feels like creating a problem for the user just for the sake of it. Every single Unix based system installs putty with the GTK+3 GUI, even Arch! Every windows user has the GUI version when installing putty as well. Universities like the one I study at uses the GTK version to manage our course enrollment as well as what many departments use to for remote connection. Why is it that Homebrew specifically cannot bundle it? There is even an attempt to build a bundle as such here #2528. Why not include that as something like putty-GTK?

**Reply by SMillerDev** [reply]
I think the reason it doesn't exist is mostly that people are just complaining about it here instead of putting the effort into fixing their issue.
